packages/react/src/components/Embed/Embed.tsx

```diff
@@ -0,0 +1,176 @@
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+
+import {
+  createShorthandFactory,
+  UIComponentProps,
+  applyAccessibilityKeyHandlers,
+  commonPropTypes,
+  AutoControlledComponent,
+} from '../../lib'
+import { embedBehavior } from '../../lib/accessibility'
+import { Accessibility, AccessibilityActionHandlers } from '../../lib/accessibility/types'
+import Icon, { IconProps } from '../Icon/Icon'
+import Image, { ImageProps } from '../Image/Image'
+import Video, { VideoProps } from '../Video/Video'
+import Box from '../Box/Box'
+import { ComponentEventHandler, ReactProps, ShorthandValue } from '../../types'
+
+export interface EmbedSlotClassNames {
+  control: string
+}
+
+export interface EmbedProps extends UIComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default embedBehavior
+   * */
+  accessibility?: Accessibility
+
+  /** Whether the embedded object should be active. */
+  active?: boolean
+
+  /** Whether the embedded object should start active. */
+  defaultActive?: boolean
+
+  /** Shorthand for an control. */
+  control?: ShorthandValue<IconProps>
+
+  /** Shorthand for an embedded iframe. */
+  iframe?: ShorthandValue
+
+  /**
+   * Called when is clicked.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All item props.
+   */
+  onClick?: ComponentEventHandler<EmbedProps>
+
+  /** Image source URL for when video isn't playing. */
+  placeholder?: ShorthandValue<ImageProps>
+
+  /** Shorthand for an embedded video. */
+  video?: ShorthandValue<VideoProps>
+}
+
+export interface EmbedState {
+  active: boolean
```

Comment:
Let's add `onActiveChanged` prop for this component, and invoke it when we change the active prop, as it will be required when we want to use it in controlled mode (see for example the `onOpenChanged` in the `Popup` component)

Reply:
Implemented as `onActiveChanged`, however I want address an inconsistency there separately because `Popup` has `onOpenChange` (without _d_ on end).

Reply:
I would expect the _d_ in the end, but agree, let's address in a separate PR, to avoid unnecessary breaking changes.